py/server/deephaven/experimental/s3.py

```diff
@@ -187,6 +198,11 @@ def __init__(self,
         except Exception as e:
             raise DHError(e, "Failed to build S3 instructions") from e
 
+    def throw_multiple_credentials_error(self,
+                    credentials1: str,
+                    credentials2: str):
+        raise DHError(f"Only one set of credentials can be set, but found {credentials1} and {credentials2}")
```

Comment:
If left here, this method should be made private -- prefixed with `_`.  The potentially better solution would be to define this function inside of the `__init__` function.